src/components/Modal/BaseModal.js

```diff
@@ -30,17 +30,28 @@ class BaseModal extends PureComponent {
         this.hideModal = this.hideModal.bind(this);
     }
 
+    componentDidMount() {
+        if (!this.props.isVisible) { return; }
+
+        // To handle closing any modal already visible when this modal is mounted, i.e. PopoverReportActionContextMenu
+        Modal.setCloseModal(this.props.onClose);
+    }
+
     componentDidUpdate(prevProps) {
         if (prevProps.isVisible === this.props.isVisible) {
             return;
         }
 
         Modal.willAlertModalBecomeVisible(this.props.isVisible);
+        Modal.setCloseModal(this.props.isVisible ? this.props.onClose : null);
```

Comment:
@situchan Do you recall why this call is done in that way? i.e. why we need to set the callback to null if this modal is not visible? Why we don't do it the same as `componentDidMount`:

```js
if (this.props.isVisible) {
    Modal.setModalClose(this.props.onClose);
}
```

Reply:
We pass `onClose` callback to `setCloseModal` to allow other parts of the app to close this modal manually.
If already closed, why we still need to have this stale callback? Also see below code why it's needed:
https://github.com/Expensify/App/blob/e48121cb35052e958928c002d2b3a7d78a930ff3/src/libs/actions/Modal.ts#L17-L25
Btw, does that cause any issue?

Reply:
> If already closed, why we still need to have this stale callback?

We may have other modals that did set the callback. If we render a new modal invisible, the old visible modal close callback will be overwritten.

Reply:
> Btw, does that cause any issue?

Not exactly, but the functional rewrite of the modal seems to be the cause of the regression. I think there was a misunderstanding of how the code is supposed to be executed. Here at this point, `componentDidMount` and `componentDidUpdate` call `Modal.setModalClose` differently (as explained above) but after the functional migration we only use `useEffect` and both logic get merged to look like `componentDidUpdate`,